### ISSUE TITLE
Request: Support absolute pathing for variable files referenced in `--terraform-var-file`?

### ISSUE THREAD

**Issue by mdaize** [infracost/infracost#2166]
We've run into a bit of a conundrum when trying to feed `infracost` with variable files with `--terraform-var-file` argument. In particular, the fact that it only supports _relative_ pathing has presented a bit of an issue.

# Context

We have a number of Terraform modules of varying designs (some with sub-directories). For example:
* RepoA might have TF files at it's root `/*.tf`
  ```
  repoA/
  |- main.tf
  |- README.md
  `- ...
  ```
* RepoB might have TF files under subdirectories `/terraform/**/*.tf`
  ```
  repoB/
  |- terraform
  |  |- main.tf
  |  `- ...
  |- README.md
  `- ...
  ```
With these particular patterns, we can readily set the `--path` to the desired directory dynamically based on pipeline parameters.

The issue presents itself since we also have a separate configuration repository like so:
  ```
  configurationRepo/
  |- env_config.tfvars
  |- ...
  `- README.md
  ```

# The Challenge

Within our CI/CD pipeline (Azure DevOps), the agent pulls both repositories (TF & Configuration) into the working path. For example:
```
agentWorkingDirectory/
|- terraform_src
|  `- main.tf (or subdir containing it)
`- config_src
   `- env_config.tfvars
```

Now, when we run `infracost breakdown --path <baseDir>`, we try to add `--terraform-var-file` to the CLI call but we use `$(Build.SourcesDirectory)` (an ADO pipeline systems variable) in the argument to reference `agentWorkingDirectory/config_src/*.tfvars` since the `terraform_src/*` could have a dynamic structure.

The resulting path that infracost see is: 
`repoN/terraform_src/azp/agent/_work/1/s/config_src/env_config.tfvars`

As you cna see, it pre-pended the variable path with the value of `--path`, then appended the value from `$(Build.SourcesDirectory)` which is an absolute path, effectively preventing dynamic patterns for using a static variable reference (outside its path).

# Possible solution?

This problem could be entirely avoidable if the `--terraform-var-files` supported absolute paths (eg. only when the value provided starts with `/`).

Any insight you might have would be quite helpful!

Thanks in advance.

**Comment by alikhajeh1** [comment]
@mdaize thanks for the awesome write-up and context!

@aliscott any reason why we can't or wouldn't want to support absolute paths for `--terraform-var-file` and `--usage-file`?

**Comment by alikhajeh1** [comment]
@mdaize I doubt it helps for your use-case but there's also a [config-file](https://www.infracost.io/docs/features/config_file/) in case you haven't seen that.

**Comment by alikhajeh1** [comment]
@mdaize thanks for the context and trying the workaround, yeah we should support absolute paths, I've added this to the project board in case others from the community have time to work on this too.

**Comment by hugorut** [comment]
Hello @mdaize  the fix for this issue has been released as part of the [v0.10.17](https://github.com/infracost/infracost/releases/tag/v0.10.17) infracost CLI. Thanks for bringing this to our attention.
